modules/standard/Types.chpl

```diff
@@ -436,53 +571,29 @@ proc isReal(e)      param  return isRealValue(e);
 /* Returns `true` if the argument is a string or the `string` type. */
 proc isString(e)    param  return isStringValue(e);
+/* Returns `true` if the argument is a bytes or the `bytes` type. */
+proc isBytes(e)     param  return isBytesValue(e);
 /* Returns `true` if the argument is an `enum` type or value, of any width. */
 proc isEnum(e)      param  return isEnumValue(e);
-/* Returns `true` if the argument is a tuple type or value. */
-proc isTuple(e)     param  return isTupleValue(e);
-/* Returns `true` if the argument is a homogeneous tuple.
-   The argument must be a tuple or any type. */
-proc isHomogeneousTuple(e: _tuple)  param  return isHomogeneousTupleValue(e);
 /* Returns `true` if the argument is a class type or value
    that is not an ``extern`` class, or when the argument is ``nil``. */
 proc isClass(e)     param  return isClassValue(e);
-/* Returns `true` if the argument is a generic type, and `false` otherwise. */
-proc isGeneric(e)   param  return false;
-/* Returns `true` if the argument is an ``owned`` class type. */
-pragma "no borrow convert"
-proc isOwnedClass(e)     param  return isOwnedClassValue(e);
-/* Returns `true` if the argument is a ``shared`` class type. */
-pragma "no borrow convert"
-proc isSharedClass(e)     param  return isSharedClassValue(e);
-/* Returns `true` if the argument is a ``unmanaged`` class type. */
-pragma "no borrow convert"
-proc isUnmanagedClass(e)     param  return isUnmanagedClassValue(e);
-/* Returns `true` if the argument is a ``borrowed`` class type. */
-pragma "no borrow convert"
-proc isBorrowedClass(e)     param  return isBorrowedClassValue(e);
-
-/* Returns `true` if the argument is a class type that can store ``nil``. */
-proc isNilableClass(e)     param  return isNilableClassValue(e);
-/* Returns `true` if the argument is a class type that cannot store ``nil``. */
-proc isNonNilableClass(e)  param  return isNonNilableClassValue(e);
-
-
 /* Returns `true` if the argument is a record type or value. */
 proc isRecord(e)    param  return isRecordValue(e);
 /* Returns `true` if the argument is a union type or value. */
 proc isUnion(e)     param  return isUnionValue(e);
+/* Returns `true` if the argument is a tuple type or value. */
+proc isTuple(e)     param  return isTupleValue(e);
 /* Returns `true` if the argument is a range type or value. */
 proc isRange(e)     param  return isRangeValue(e);
-/* Returns `true` if the argument is a domain map or a domain map type. */
-proc isDmap(e)      param  return isDmapValue(e);
 /* Returns `true` if the argument is a domain or a domain type. */
 proc isDomain(e)    param  return isDomainValue(e);
 /* Returns `true` if the argument is an array or an array type. */
 proc isArray(e)     param  return isArrayValue(e);
+/* Returns `true` if the argument is a domain map or a domain map type. */
+proc isDmap(e)      param  return isDmapValue(e);
 pragma "no doc"
 proc isSync(e: sync)     param  return true; // workaround: not isSyncValue
```

Comment:
Is this work-around still necessary?

Reply:
I must of missed this earlier.  If I try changing this to this (I assume the workaround referred to is having the extra overloads):

```chpl
/* Returns ``true`` if the argument is a ``sync`` type or a ``sync`` variable. */
proc isSync(e)      param  return isSyncValue(e);
/* Returns ``true`` if the argument is a ``single`` type or a ``single`` variable. */
proc isSingle(e)    param  return isSingleValue(e);
```

I'm  able to compile and run this little test program just fine:

```chpl
var a : sync int = 1;
var b : int = 2;
var c : single int = 3;
var d : int = 4;

writeln("isSync(a) = ", isSync(a));
writeln("isSyncValue(a) = ", isSyncValue(a));
writeln("isSync(b) = ", isSync(b));
writeln("isSyncValue(b) = ", isSyncValue(b));
writeln();
writeln("isSingle(c) = ", isSingle(c));
writeln("isSingleValue(c) = ", isSingleValue(c));
writeln("isSingle(d) = ", isSingle(d));
writeln("isSingleValue(d) = ", isSingleValue(d));
```

And the output is what you'd expect if you run it:

```
isSync(a) = true
isSyncValue(a) = true
isSync(b) = false
isSyncValue(b) = false

isSingle(c) = true
isSingleValue(c) = true
isSingle(d) = false
isSingleValue(d) = false
```

So I guess the answer is "no" the work-around is not necessary. I'll go ahead and pass this through paratests on the off-chance I missing something but it seems like it should go through fine.

Reply:
Might have been due to having them in separate files or something weird.  In any case, if that passes testing, great!